optuna/samplers/nsgaiii.py

```diff
@@ -0,0 +1,634 @@
+            `target` points since the algorithm prioritizes individuals around reference points.
+
+        dividing_parameter:
+            An parameter to determine the density of default reference points. This parameter
+            determines how many divisions are made between reference points on each axis. The
+            smaller this value is, the less reference points you have. The default value is 3.
+            Note that this parameter is not used when reference_points is not None.
+
+            .. note::
+                Other parameters are the same as :class:`~optuna.samplers.nsgaii.NSGAIISampler`.
+
+    """
+
+    def __init__(
+        self,
+        population_size: int = 50,
+        reference_points: np.ndarray | None = None,
+        dividing_parameter: int = 3,
+        mutation_prob: float | None = None,
+        crossover: BaseCrossover | None = None,
+        crossover_prob: float = 0.9,
+        swapping_prob: float = 0.5,
+        seed: int | None = None,
+        constraints_func: Callable[[FrozenTrial], Sequence[float]] | None = None,
+    ) -> None:
+        # TODO(ohta): Reconsider the default value of each parameter.
+
+        if not isinstance(population_size, int):
+            raise TypeError("`population_size` must be an integer value.")
+
+        if population_size < 2:
+            raise ValueError("`population_size` must be greater than or equal to 2.")
+
+        if not (mutation_prob is None or 0.0 <= mutation_prob <= 1.0):
+            raise ValueError(
+                "`mutation_prob` must be None or a float value within the range [0.0, 1.0]."
+            )
+
+        if not (0.0 <= crossover_prob <= 1.0):
+            raise ValueError("`crossover_prob` must be a float value within the range [0.0, 1.0].")
+
+        if not (0.0 <= swapping_prob <= 1.0):
+            raise ValueError("`swapping_prob` must be a float value within the range [0.0, 1.0].")
+
+        if constraints_func is not None:
+            warnings.warn(
+                "The constraints_func option is an experimental feature."
+                " The interface can change in the future.",
+                ExperimentalWarning,
+            )
+
+        if crossover is None:
+            crossover = UniformCrossover(swapping_prob)
+        if not isinstance(crossover, BaseCrossover):
+            raise ValueError(
+                f"'{crossover}' is not a valid crossover."
+                " For valid crossovers see"
+                " https://optuna.readthedocs.io/en/stable/reference/samplers.html."
+            )
+
+        if population_size < crossover.n_parents:
+            raise ValueError(
+                f"Using {crossover},"
+                f" the population size should be greater than or equal to {crossover.n_parents}."
+                f" The specified `population_size` is {population_size}."
+            )
+
+        self.reference_points = reference_points
```

Comment:
Is this attribute intended to be referenced by users? If not, how about starting the name from `_`?

Reply:
I see. My personal opinion is that if you want the user to be able to change this attribute, and they can, you should implement a setter function and a getter function. setter/getter implementation is not that much work, so you can use this PR Would you do it? By the way, in any case, I think it is better to start the attribute name itself with `_`.

Reply:
Thank you for your advice. I privatized `reference_points` and defined a getter and a setter.